azurerm/internal/services/redis/resource_arm_redis_linked_server.go

```diff
@@ -0,0 +1,236 @@
+		}
+
+		log.Printf("[DEBUG] Waiting for Redis Linked Server (%s) to become available", d.Get("name"))
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"Linking", "Updating", "Creating", "Syncing"},
+			Target:     []string{"Succeeded"},
+			Refresh:    redisLinkedServerStateRefreshFunc(ctx, client, resourceGroup, redisCacheName, name),
+			MinTimeout: 15 * time.Second,
+			Timeout:    d.Timeout(schema.TimeoutCreate),
+		}
+
+		if _, err = stateConf.WaitForState(); err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Error waiting for Redis Linked Server (%s) to become available: %s", d.Get("name"), err))
+		}
+
+		d.SetId(*read.ID)
+
+		return resource.NonRetryableError(resourceArmRedisLinkedServerRead(d, meta))
+	})
+}
+
+func resourceArmRedisLinkedServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Redis.LinkedServerClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	redisCacheName := id.Path["Redis"]
+	name := id.Path["linkedServers"]
+
+	resp, err := client.Get(ctx, resourceGroup, redisCacheName, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Redis Linked Server %q was not found in Cache %q / Resource Group %q - removing from state", name, redisCacheName, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure Redis Linked Server %q: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("redis_cache_name", redisCacheName)
+	d.Set("resource_group_name", resourceGroup)
+	if props := resp.LinkedServerProperties; props != nil {
+		d.Set("linked_redis_cache_id", props.LinkedRedisCacheID)
+		d.Set("linked_redis_cache_location", props.LinkedRedisCacheLocation)
+		d.Set("server_role", string(props.ServerRole))
+	}
+
+	return nil
+}
+
+func resourceArmRedisLinkedServerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Redis.LinkedServerClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	redisCacheName := id.Path["Redis"]
+	name := id.Path["linkedServers"]
+
+	resp, err := client.Delete(ctx, resourceGroup, redisCacheName, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp) {
+			return fmt.Errorf("Error issuing AzureRM delete request of Redis Linked Server %q (cache %q / resource group %q): %+v", name, redisCacheName, resourceGroup, err)
+		}
+	}
+
+	// No LinkedServerDeleteFuture
+	// https://github.com/Azure/azure-sdk-for-go/issues/12159
+
+	for {
+		time.Sleep(1 * time.Minute)
+		resp, err := client.Get(ctx, resourceGroup, redisCacheName, name)
```

Comment:
Could we use a wait for state function like above as this could never finish

Reply:
Yes definitely.  I was confused.  I misread a comment in Slack and thought I needed futures for this.  I reread the comment.  Now I understand.  This would be a better solution.  Testing it now.